src/compiler/program.ts

```diff
@@ -1890,8 +1890,11 @@ namespace ts {
             for (const node of file.statements) {
                 collectModuleReferences(node, /*inAmbientModule*/ false);
             }
-            if ((file.flags & NodeFlags.PossiblyContainsDynamicImport) || isJavaScriptFile) {
-                collectDynamicImportOrRequireCalls(file);
+            if ((file.flags & NodeFlags.PossiblyContainsDynamicImport)) {
+                collectDynamicImports();
+            }
+            if (isJavaScriptFile) {

```

Comment:
I think we can still keep this a single pass by passing regexp instead.

Reply:
It's possible, but IMO not very useful. It would save around 4 lines of code just to add new code to get determine the correct regexp to use. It also adds complexity by moving the regexp out of the function where its matches are used.
I doubt having a single pass of a complex regexp outperforms two passes with a simple regex each (also considering potential inlining).

Reply:
It will save pass I think.